Accessibility statement

mtthwhggns opened the pull request:
# Description

It's a legal requirement that we show a statement of our compliance with the EU accessibility directive.

This adds the relevant templates.

No translations are done yet. 

# Testing

*Remember to add or update unit tests for new features and changes.*
*Remember to also run all tests available in the project in order to see that they still pass.*

*Describe here what needs to be tested MANUALLY due to these changes.*
*Think of what is affected by these changes and could become broken (and can not be easily tested automatically)?*

# Have you updated the README or other relevant documentation?

*Don't forget to update the documentation if it is relevant for this PR.*

# Is it [Done](https://wiki.aalto.fi/display/EDIT/Definition+of+Done)?

*Clean up your git commit history before submitting the pull request!*

- [ ] I (developer) have created unit tests and the tests pass
- [ ] I (developer) have created functional tests (Selenium tests) if applicable
- [ ] I (developer) have tested the changes manually
- [ ] Reviewer has finished the code review
- [ ] After the review, the developer has made changes accordingly
- [ ] Customer/Teacher has accepted the implementation of the feature

raphendyr commented on userprofile/views.py:
```diff
@@ -128,5 +128,44 @@ def get_common_objects(self):
         self.privacy_text = privacy_text
         self.note("privacy_text")
 
+class AccessibilityStatementView(UserProfileView):
```
Can you add extra newlines before and after the class (i.e. 2 lines between different classes.

raphendyr commented on userprofile/views.py:
```diff
@@ -128,5 +128,44 @@ def get_common_objects(self):
         self.note("privacy_text")
 
+class AccessibilityStatementView(UserProfileView):
+    access_mode=ACCESS.ANONYMOUS
```
Even though this is a copy-paste, can you add spaces around `=`.

raphendyr commented on userprofile/views.py:
```diff
@@ -128,5 +128,44 @@ def get_common_objects(self):
+        if not accessibility_statement:
+            local_template_name = "institution_accessibility_text{}.html"
+            local_template = try_get_template(local_template_name.format(lang))
+            if not local_template and len(lang) > 3:
```
This logic should be refactored to a function for all of the 3 uses (1 in privacy notice and 2 times here), but I think it can be left for later. Can you add a comment `# TODO: refactor to a helper function`?

mtthwhggns commented:
Update on the status of this: It currently contains a statement where I've swapped out the name of the University for "we" and "us" - so that others can use this as a template if their institution isn't providing one.

The Aalto-specific template has moved elsewhere.

From my point of view, I think I've done everything suggested on this PR.

Mankro commented on locale/fi/LC_MESSAGES/django.po:
```diff
@@ -3697,6 +3697,14 @@ msgstr ""
+
+#: userprofile/views.py:153
+msgid "No local accessibility statement. Please notify the site's owner!"
+msgstr ""
```
"No local accessibility statement. Please notify the site's owner!"
"Ei paikallista saavutettavuusselostetta. Ilmoitathan tästä sivuston ylläpitäjälle!"

Mankro commented on locale/fi/LC_MESSAGES/django.po:
```diff
@@ -3697,6 +3697,14 @@ msgstr ""
+msgstr ""
+
+#: userprofile/views.py:165
+msgid "No system-wide accessibility statement found."
```
"No system-wide accessibility statement found."
"Järjestelmätason saavutettavuusselostetta ei löytynyt."

Mankro reviewed (changes requested):
It's good. Would you add the two small translations in django.po that I commented on?

mtthwhggns commented:
> It's good. Would you add the two small translations in django.po that I commented on?
> Edit: you could rebase and squash then.

Oh, yes! I forgot about these.

I've added them now, squashed and rebased.

Mankro commented on userprofile/views.py:
```diff
@@ -128,5 +128,47 @@ def get_common_objects(self):
+
+            system_accessibility_statement = system_template.render() if system_template else _("No system-wide accessibility statement found.")
+
+            accessibility_statement = f'{local_accessibility_statement}{system_accessibility_statement}'
```
Is this the new Python string formatting syntax that is only available from Python 3.6 onwards? Our production server is still on Python 3.5 and we must therefore remain compatible with Python 3.5.

mtthwhggns replied:
Ok. 

I assumed the docket container was using the same version as production. 

I switched to this because concatenation of the two strings with `+` fails if the file is missing. 

@Mankro What's the right way to do this in our version of Python?

mtthwhggns replied:
I'm no Python expert, but this seems to work: 

```
str(local_accessibility_statement) + str(system_accessibility_statement
```

Mankro replied:
`_` refers to the function `ugettext_lazy`, but we don't need lazy translation in this case. The template `render` method does not return lazy strings, either. I think you can call `gettext` instead of `_` here and then the string concatenate `+` operator will work directly. You need to import `gettext` to use it. Note that Django has the `format_lazy` function for doing string formatting on lazy strings.

mtthwhggns replied:
I've done this now.

Mankro commented:
@mtthwhggns did you notice yet that the git commit message got broken?

mtthwhggns commented:
> @mtthwhggns did you notice yet that the git commit message got broken?

I didn't, thanks. Fixed now.
